src/wordfish.cpp

```diff
@@ -171,19 +206,20 @@ Rcpp::List wordfishcpp(SEXP wfm, SEXP dir, SEXP priors, SEXP tol, SEXP disp, SEX
         }	  			
         
         alpha = alpha - mean(alpha);
-        theta = (theta - mean(theta))/sd(theta);		
+        theta = (theta - mean(theta))/stddev(theta);		
         
         // CHECK LOG-POSTERIOR FOR CONVERGENCE
         lastlp = lp;
-        lp = -1.0*(sum(0.5 * ((alpha*alpha)*(priorprecalpha))) + sum(0.5 * ((psi*psi)*(priorprecpsi))) + sum(0.5 * ((beta*beta)*(priorprecbeta))) + sum(0.5 * ((theta*theta)*(priorprectheta))));
-        for (int i=0; i < N; i++){
-            for (int k=0; k < K; k++){
+        lp = -1.0*(accu(0.5 * ((alpha % alpha) * priorprecalpha)) + accu(0.5 * ((psi % psi) * priorprecpsi))
+                       + accu(0.5 * ((beta % beta) * priorprecbeta)) + accu(0.5 * ((theta % theta) * priorprectheta)));
+        for (std::size_t i=0; i < N; i++){
+            for (std::size_t k=0; k < K; k++){
                 loglambdaik = alpha(i) + psi(k) + beta(k)*theta(i);
-                lp = lp + loglambdaik*Y(i,k)-exp(loglambdaik);
+                lp = lp + loglambdaik*wfm(i,k)-exp(loglambdaik);
             }
         }
-        // Rprintf("%d: %f2\\n",outeriter,lp);
-        
+        Rcout<<"outeriter="<<outeriter<<"  lp - lastlp= "<<lp - lastlp<<std::endl;
```

Comment:
I just remembered I kept that line for you and @conjugateprior to review the convergence.